sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -67,64 +65,63 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
   private abortControllers: Map<string, AbortController[] | undefined>;
 
   constructor(options?: TokenCredentialOptions) {
-    super(
-      undefined,
-      createPipelineFromOptions({
-        ...options,
-        deserializationOptions: {
-          expectedContentTypes: {
-            json: ["application/json", "text/json", "text/plain"]
-          }
-        }
-      })
-    );
+    const packageDetails = `azsdk-js-identity/2.0.0-beta.5`;
+    const userAgentPrefix = options?.userAgentOptions?.userAgentPrefix
+      ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
+      : `${packageDetails}`;
 
-    this.baseUri = this.authorityHost = getIdentityClientAuthorityHost(options);
-
-    if (!this.baseUri.startsWith("https:")) {
+    const baseUri = getIdentityClientAuthorityHost(options);
+    if (!baseUri.startsWith("https:")) {
       throw new Error("The authorityHost address must use the 'https' protocol.");
     }
 
-    this.abortControllers = new Map();
-  }
+    super({
+      requestContentType: "application/json; charset=utf-8",
+      ...options,
+      userAgentOptions: {
+        userAgentPrefix
+      },
+      baseUri
+    });
 
-  createWebResource(requestOptions: RequestPrepareOptions): WebResource {
-    const webResource = new WebResource();
-    webResource.prepare(requestOptions);
-    return webResource;
+    this.authorityHost = baseUri;
+    this.abortControllers = new Map();
   }
 
   async sendTokenRequest(
-    webResource: WebResource,
+    request: PipelineRequest,
     expiresOnParser?: (responseBody: any) => number
   ): Promise<TokenResponse | null> {
-    logger.info(`IdentityClient: sending token request to [${webResource.url}]`);
-    const response = await this.sendRequest(webResource);
+    logger.info(`IdentityClient: sending token request to [${request.url}]`);
+    const response = await this.sendRequest(request);
 
     expiresOnParser =
       expiresOnParser ||
       ((responseBody: any) => {
         return Date.now() + responseBody.expires_in * 1000;
       });
 
-    if (response.status === 200 || response.status === 201) {
+    if (response.bodyAsText && (response.status === 200 || response.status === 201)) {
+      const parsedBody = parse<{
+        token?: string;
+        access_token?: string;
+        refresh_token?: string;
+      }>(response.bodyAsText);
+
       const token = {
         accessToken: {
-          token: response.parsedBody.access_token,
-          expiresOnTimestamp: expiresOnParser(response.parsedBody)
+          token: parsedBody.token ?? parsedBody.access_token!,
```

Comment:
Assert-defined here makes me nervous, since it wasn't present before. What changed that makes us require the `!`?

Also, isn't this a semantic change as well? We're going from only looking for `access_token` to looking for `token` first, then `access_token`. Why the change?

Reply:
We had some tests passing a `token` property in. I believe I could have missed this logic the first time I did the migration. I’ll check other languages just in case.

Reply:
After checking other languages, access_token is the only one I should consider. I’ll fix this